internal/resources/app_endpoint.go

```diff
@@ -419,11 +419,20 @@ func (a *AppEndpoint) Read(ctx context.Context, req resource.ReadRequest, resp *
 		return
 	}
 
-	var organizationId = state.OrganizationId.ValueString()
-	var projectId = state.ProjectId.ValueString()
-	var clusterId = state.ClusterId.ValueString()
-	var appServiceId = state.AppServiceId.ValueString()
-	var endpointName = state.Name.ValueString()
+	IDs, err := state.Validate()
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error refreshing app endpoint",
```

Comment:
nit

this is misleading error summary.  it's validation error, not refresh error